src/test/a11y/customChecks.ts

```diff
@@ -0,0 +1,190 @@
+      if (link) {
+        expect('Change ' + dt).to.equal(link.textContent.trim())
+      } else {
+        console.log('INFO: no change link available for "' + dt + '"')
+      }
+    }
+  }
+}
+
+/**
+ *
+ * @param window
+ * @param document
+ * expect error container with class 'error-summary' is present
+ * expect h2 tag for error messages
+ * expect 'There was a problem' text in h2
+ * only one h1 tag per page
+ */
+export const checkError = (window: Window, document: Document) => {
+  const className = 'error-summary'
+  const errorContainer = document.getElementsByClassName(className)
+  // should contain only one error-summary
+  expect(errorContainer.length, `div element with '${className}' must be present`).to.be.equal(1)
+  // error-summary div should contain <h2>There was a problem</h2>
+  const heading = errorContainer[0].getElementsByTagName('h2')
+  expect(heading.length, 'Error heading should use h2 tag').to.be.eq(1)
+  expect(heading[0].textContent.trim()).to.be.eq('There was a problem')
+}
+
+export const checkRole = (window: Window, document: Document) => {
+  // this expects and validates 'Total monthly income' 'p' tag has role="status" and aria-live="polite"
+  /**
+   * <p class="heading-small calculation-outcome-container" role="status" aria-live="polite">
+   *  {{ t('Total monthly income:') }} £<span class="total-monthly-income-expense"> {{ totalMonthlyIncomeExpense | default('0.00') }} </span>
+   * </p>
+   */
+  const definitionList = document.getElementsByClassName('calculation-outcome-container')
+  expect(definitionList.length, 'Income / Expence page must have role="status" and aria-live="polite" for the "p" tag').to.be.greaterThan(0)
+
+  for (let i = 0; i < definitionList.length; i++) {
+    const attributesName = definitionList[i].getAttributeNames()
+    expect(attributesName.length,'Total Income / Expence "p" tag must have "role" & "aria-live" attributes').to.be.greaterThan(2)
+    expect(attributesName).to.contains('role')
+    expect(attributesName).to.contains('aria-label')
+    if (attributesName.length >= 3) {
+      expect(definitionList[0].getAttribute('role')).to.equal('status')
+      expect(definitionList[0].getAttribute('aria-live')).to.equal('polite')
+    } else {
+      console.log('INFO: no change role and aria-live attributes present in the "p" tag')
+    }
+  }
+}
+
+export const checkButton = (window: Window, document: Document) => {
+  // this expects and validates 'aria-label' attribute is present in the submit button"
+  /**
+   * <input type="submit" class="button" aria-label="I confirm I’ve read this information about resolving disputes" value="I confirm I’ve read this">
+   */
+  const definitionList = document.getElementsByClassName('button')
+  expect(definitionList.length, 'resolving-this-dispute page must have submite button').to.be.greaterThan(0)
+
+  for (let i = 0; i < definitionList.length; i++) {
```

Comment:
I don't think we need for loop here. Since we always expect 1 input button

Reply:
removed the loop 
![image](https://user-images.githubusercontent.com/41957682/91153262-e692b780-e6b7-11ea-9e9c-2a8563f0f453.png)